File: app/src/main/java/com/github/gfx/helium/di/FragmentComponent.java
package com.github.gfx.helium.di;

import com.github.gfx.helium.di.scope.FragmentScope;
import com.github.gfx.helium.fragment.EpitomeEntryFragment;
import com.github.gfx.helium.fragment.HatebuEntryFragment;
import com.github.gfx.helium.fragment.SettingsFragment;
import com.github.gfx.helium.fragment.TimelineFragment;

import dagger.Subcomponent;

@FragmentScope
@Subcomponent(modules = {FragmentModule.class})
public interface FragmentComponent {

    void inject(HatebuEntryFragment fragment);

    void inject(EpitomeEntryFragment fragment);

    void inject(TimelineFragment fragment);

    void inject(SettingsFragment fragment);

}

File: app/src/main/java/com/github/gfx/helium/di/FragmentComponent.kt
package com.github.gfx.helium.di

import com.github.gfx.helium.di.scope.FragmentScope
import com.github.gfx.helium.fragment.EpitomeEntryFragment
import com.github.gfx.helium.fragment.HatebuEntryFragment
import com.github.gfx.helium.fragment.SettingsFragment
import com.github.gfx.helium.fragment.TimelineFragment

import dagger.Subcomponent

@FragmentScope
@Subcomponent(modules = arrayOf(FragmentModule::class))
interface FragmentComponent {

    fun inject(fragment: HatebuEntryFragment)

    fun inject(fragment: EpitomeEntryFragment)

    fun inject(fragment: TimelineFragment)

    fun inject(fragment: SettingsFragment)

}
